deduction_rules.py

```diff
@@ -167,4 +167,162 @@ def line_square(sudoku):
                     r,c=local_to_global(sec,i,j)
                     if c!=col:
                         made_deduction|=sudoku.ban(r,c,val+1,"col_square",reason)
+    return made_deduction
+
+def naked_trios(sudoku):
+    """RULE: if v,w,x can be written only to three cells in this row/col/sec, then remove it from the other cells of the row/col/sec."""
```

Comment:
This is the definition of a `hidden_trio` usually; `naked_trio` should be the *dual* of this.

Reply:
I updated the dosctring. I hope it is clearer now.